### PR TITLE
Use GCC/Clang builtins for overflows checking

### LINKED ISSUE #4356: Use GCC/Clang builtins for overflows checking
See `librz/include/rz_types_overflow.h`

See https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html

An example how Meson checks for those could be added: https://github.com/rizinorg/rizin/commit/bbc5457b7a8f7362da1caacd74d634060b71deda

### PR DESCRIPTION
**Pull request by KBhushan07**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**
Used GCC builtins for overflow checking instead of custom overflow checking code .

**Closing issues**
closes #4356.

### CODE REVIEW

File: librz/include/rz_types_overflow.h [hunk 1]
```diff
@@ -8,55 +8,78 @@
...
+#define SZT_ADD_OVFCHK(x, y)  __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
+#define SSZT_ADD_OVFCHK(a, x) __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
+#define UT64_ADD_OVFCHK(x, y) __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
+#define ST64_ADD_OVFCHK(a, x) __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
+#define UT32_ADD_OVFCHK(x, y) __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
+#define ST32_ADD_OVFCHK(a, x) __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
+#define UT16_ADD_OVFCHK(x, y) __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
+#define ST16_ADD_OVFCHK(a, b) __builtin_add_overflow_p (a, b, (__typeof__ ((a) + (b))) 0)
+#define UT8_ADD_OVFCHK(x, y)  __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
+#define ST8_ADD_OVFCHK(a, x)  __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
```
Comment (wargio) [inline librz/include/rz_types_overflow.h:20]: these are not compatible with MSVC

Comment (wargio) [reply librz/include/rz_types_overflow.h:20]: if you want you can make an `#if __clang__ || __gcc__` and have the else with the old check

Comment (XVilka) [reply librz/include/rz_types_overflow.h:20]: No,  they should be checked in meson, just how it was done for bswap builtins, I referred corresponding commit in the issue

Comment (wargio) [reply librz/include/rz_types_overflow.h:20]: makes sense.

Comment (KBhushan07) [reply librz/include/rz_types_overflow.h:20]: I have made following changes.
1. In the Meson build file (meson.build), I've added checks for the availability of these built-in functions.
Screenshot from 2024-03-16 15-38-33

2.In the rz_types.h header file, I've defined macros HAVE___BUILTIN_ADD_OVERFLOW, HAVE___BUILTIN_SUB_OVERFLOW, and HAVE___BUILTIN_MUL_OVERFLOW based on the values obtained from Meson.
Screenshot from 2024-03-16 15-39-59

3. I've used conditional compilation (#ifdef) in the rz_types_overflow.h to define macros depending on whether these built-in functions are available.
Screenshot from 2024-03-16 15-40-30

Could you please review these changes and confirm whether I'm on the right track? If there are any additional steps I should take or if any adjustments are needed, please let me know.

Comment (XVilka) [reply librz/include/rz_types_overflow.h:20]: Please avoid asking via images - it makes reviewing harder. Just push whatever changes directly to the PR, if any changes will be necessary, we will request them via the review process.

Comment (XVilka) [reply librz/include/rz_types_overflow.h:20]: For defines, please use `rz_userconf.h`, it is autogenerated based on the meson detection, see how it's done in the commit I referenced.

**Review by XVilka — CHANGES_REQUESTED** [review]
Please remove `xx` file

**Comment by KBhushan07** [comment]
> Please remove `xx` file

Ohh Sorry.

File: librz/include/rz_types.h [hunk 2]
```diff
@@ -14,6 +14,7 @@
 #include <inttypes.h>
 #endif
 
+
```
Comment (XVilka) [inline librz/include/rz_types.h]: Remove this unnecessary change

File: librz/include/rz_types_overflow.h [hunk 3]
```diff
@@ -5,9 +5,21 @@
 
 // TODO: Use CLANG/GCC builtins if available: __builtin_mul_overflow
 
-// ADD
-// if ((x > 0) && (a > INT_MAX - x)) /* `a + x` would overflow */;
-// if ((x < 0) && (a < INT_MIN - x)) /* `a + x` would underflow */;
+//Using CLANG/GCC builtins
```
Comment (XVilka) [inline librz/include/rz_types_overflow.h]: Space before `Using`. Also, it's probably better to write "Using compiler builtins when available"

File: librz/include/rz_types_overflow.h [hunk 4]
```diff
@@ -5,9 +5,21 @@
 
 // TODO: Use CLANG/GCC builtins if available: __builtin_mul_overflow
```
Comment (XVilka) [inline librz/include/rz_types_overflow.h]: Remove this line since it's now solved

File: librz/include/rz_types_overflow.h [hunk 5]
```diff
@@ -32,15 +56,23 @@
 #define ST16_SUB_OVFCHK(a, b) ST16_ADD_OVFCHK(a, -(b))
 #define UT8_SUB_OVFCHK(a, b)  UT8_ADD_OVFCHK(a, -(b))
 #define ST8_SUB_OVFCHK(a, b)  ST8_ADD_OVFCHK(a, -(b))
-
+#endif
```
Comment (XVilka) [inline librz/include/rz_types_overflow.h:57]: Empty line before `// MUL`

File: librz/include/rz_types_overflow.h [hunk 6]
```diff
@@ -32,15 +56,23 @@
...
 // MUL
+
```
Comment (XVilka) [inline librz/include/rz_types_overflow.h:58]: Also this empty line is probably unnecessary

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by XVilka** [comment]
TinyCC build fails:
```
FAILED: librz/syscall/d/darwin-arm-64.sdb 
/home/runner/work/rizin/rizin/build/librz/util/sdb_native librz/syscall/d/darwin-arm-64.sdb == librz/syscall/d/darwin-arm-64.sdb.txt.tmp
/home/runner/work/rizin/rizin/build/librz/util/sdb_native: symbol lookup error: /home/runner/work/rizin/rizin/build/librz/util/librz_util.so.0.8: undefined symbol: __builtin_add_overflow_p
[688/1955] Generating librz/syscall/d/darwin-arm-32.sdb with a custom command
FAILED: librz/syscall/d/darwin-arm-32.sdb 
/home/runner/work/rizin/rizin/build/librz/util/sdb_native librz/syscall/d/darwin-arm-32.sdb == librz/syscall/d/darwin-arm-32.sdb.txt.tmp
/home/runner/work/rizin/rizin/build/librz/util/sdb_native: symbol lookup error: /home/runner/work/rizin/rizin/build/librz/util/librz_util.so.0.8: undefined symbol: __builtin_add_overflow_p
[689/1955] Generating librz/syscall/d/darwin-x86-32.sdb with a custom command
FAILED: librz/syscall/d/darwin-x86-32.sdb 
/home/runner/work/rizin/rizin/build/librz/util/sdb_native librz/syscall/d/darwin-x86-32.sdb == librz/syscall/d/darwin-x86-32.sdb.txt.tmp
/home/runner/work/rizin/rizin/build/librz/util/sdb_native: symbol lookup error: /home/runner/work/rizin/rizin/build/librz/util/librz_util.so.0.8: undefined symbol: __builtin_add_overflow_p
```
See https://github.com/rizinorg/rizin/actions/runs/8307236785/job/22736219846?pr=4363#step:9:726

**Comment by XVilka** [comment]
Apart from forementioned TinyCC linking error, formatting needs to be fixed as well:
```
librz/include/rz_types_overflow.h:10:55: error: code should be clang-formatted [-Wclang-format-violations]
#define SZT_ADD_OVFCHK(x, y)  __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                      ^
librz/include/rz_types_overflow.h:10:74: error: code should be clang-formatted [-Wclang-format-violations]
#define SZT_ADD_OVFCHK(x, y)  __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:10:87: error: code should be clang-formatted [-Wclang-format-violations]
#define SZT_ADD_OVFCHK(x, y)  __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:11:55: error: code should be clang-formatted [-Wclang-format-violations]
#define SZT_ADD_OVFCHK(x, y)  __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                      ^
librz/include/rz_types_overflow.h:11:74: error: code should be clang-formatted [-Wclang-format-violations]
#define SZT_ADD_OVFCHK(x, y)  __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:11:87: error: code should be clang-formatted [-Wclang-format-violations]
#define SZT_ADD_OVFCHK(x, y)  __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:12:55: error: code should be clang-formatted [-Wclang-format-violations]
#define SSZT_ADD_OVFCHK(a, x) __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
                                                      ^
librz/include/rz_types_overflow.h:12:74: error: code should be clang-formatted [-Wclang-format-violations]
#define SSZT_ADD_OVFCHK(a, x) __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:12:87: error: code should be clang-formatted [-Wclang-format-violations]
#define SSZT_ADD_OVFCHK(a, x) __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:13:55: error: code should be clang-formatted [-Wclang-format-violations]
#define UT64_ADD_OVFCHK(x, y) __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                      ^
librz/include/rz_types_overflow.h:13:74: error: code should be clang-formatted [-Wclang-format-violations]
#define UT64_ADD_OVFCHK(x, y) __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:13:87: error: code should be clang-formatted [-Wclang-format-violations]
#define UT64_ADD_OVFCHK(x, y) __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:14:55: error: code should be clang-formatted [-Wclang-format-violations]
#define ST64_ADD_OVFCHK(a, x) __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
                                                      ^
librz/include/rz_types_overflow.h:14:74: error: code should be clang-formatted [-Wclang-format-violations]
#define ST64_ADD_OVFCHK(a, x) __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:14:87: error: code should be clang-formatted [-Wclang-format-violations]
#define ST64_ADD_OVFCHK(a, x) __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:15:55: error: code should be clang-formatted [-Wclang-format-violations]
#define UT32_ADD_OVFCHK(x, y) __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                      ^
librz/include/rz_types_overflow.h:15:74: error: code should be clang-formatted [-Wclang-format-violations]
#define UT32_ADD_OVFCHK(x, y) __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:15:87: error: code should be clang-formatted [-Wclang-format-violations]
#define UT32_ADD_OVFCHK(x, y) __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:16:55: error: code should be clang-formatted [-Wclang-format-violations]
#define ST32_ADD_OVFCHK(a, x) __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
                                                      ^
librz/include/rz_types_overflow.h:16:74: error: code should be clang-formatted [-Wclang-format-violations]
#define ST32_ADD_OVFCHK(a, x) __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:16:87: error: code should be clang-formatted [-Wclang-format-violations]
#define ST32_ADD_OVFCHK(a, x) __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:17:55: error: code should be clang-formatted [-Wclang-format-violations]
#define UT16_ADD_OVFCHK(x, y) __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                      ^
librz/include/rz_types_overflow.h:17:74: error: code should be clang-formatted [-Wclang-format-violations]
#define UT16_ADD_OVFCHK(x, y) __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:17:87: error: code should be clang-formatted [-Wclang-format-violations]
#define UT16_ADD_OVFCHK(x, y) __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:18:55: error: code should be clang-formatted [-Wclang-format-violations]
#define ST16_ADD_OVFCHK(a, b) __builtin_add_overflow_p (a, b, (__typeof__ ((a) + (b))) 0)
                                                      ^
librz/include/rz_types_overflow.h:18:74: error: code should be clang-formatted [-Wclang-format-violations]
#define ST16_ADD_OVFCHK(a, b) __builtin_add_overflow_p (a, b, (__typeof__ ((a) + (b))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:18:87: error: code should be clang-formatted [-Wclang-format-violations]
#define ST16_ADD_OVFCHK(a, b) __builtin_add_overflow_p (a, b, (__typeof__ ((a) + (b))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:19:55: error: code should be clang-formatted [-Wclang-format-violations]
#define UT8_ADD_OVFCHK(x, y)  __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                      ^
librz/include/rz_types_overflow.h:19:74: error: code should be clang-formatted [-Wclang-format-violations]
#define UT8_ADD_OVFCHK(x, y)  __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:19:87: error: code should be clang-formatted [-Wclang-format-violations]
#define UT8_ADD_OVFCHK(x, y)  __builtin_add_overflow_p (x, y, (__typeof__ ((x) + (y))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:20:55: error: code should be clang-formatted [-Wclang-format-violations]
#define ST8_ADD_OVFCHK(a, x)  __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
                                                      ^
librz/include/rz_types_overflow.h:20:74: error: code should be clang-formatted [-Wclang-format-violations]
#define ST8_ADD_OVFCHK(a, x)  __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:20:87: error: code should be clang-formatted [-Wclang-format-violations]
#define ST8_ADD_OVFCHK(a, x)  __builtin_add_overflow_p (a, x, (__typeof__ ((a) + (x))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:36:55: error: code should be clang-formatted [-Wclang-format-violations]
#define SZT_SUB_OVFCHK(a, b)  __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                      ^
librz/include/rz_types_overflow.h:36:74: error: code should be clang-formatted [-Wclang-format-violations]
#define SZT_SUB_OVFCHK(a, b)  __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:36:87: error: code should be clang-formatted [-Wclang-format-violations]
#define SZT_SUB_OVFCHK(a, b)  __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:37:55: error: code should be clang-formatted [-Wclang-format-violations]
#define SSZT_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                      ^
librz/include/rz_types_overflow.h:37:74: error: code should be clang-formatted [-Wclang-format-violations]
#define SSZT_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:37:87: error: code should be clang-formatted [-Wclang-format-violations]
#define SSZT_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:38:55: error: code should be clang-formatted [-Wclang-format-violations]
#define UT64_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                      ^
librz/include/rz_types_overflow.h:38:74: error: code should be clang-formatted [-Wclang-format-violations]
#define UT64_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:38:87: error: code should be clang-formatted [-Wclang-format-violations]
#define UT64_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:39:55: error: code should be clang-formatted [-Wclang-format-violations]
#define ST64_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                      ^
librz/include/rz_types_overflow.h:39:74: error: code should be clang-formatted [-Wclang-format-violations]
#define ST64_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:39:87: error: code should be clang-formatted [-Wclang-format-violations]
#define ST64_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:40:55: error: code should be clang-formatted [-Wclang-format-violations]
#define UT32_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                      ^
librz/include/rz_types_overflow.h:40:74: error: code should be clang-formatted [-Wclang-format-violations]
#define UT32_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:40:87: error: code should be clang-formatted [-Wclang-format-violations]
#define UT32_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:41:55: error: code should be clang-formatted [-Wclang-format-violations]
#define ST32_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                      ^
librz/include/rz_types_overflow.h:41:74: error: code should be clang-formatted [-Wclang-format-violations]
#define ST32_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:41:87: error: code should be clang-formatted [-Wclang-format-violations]
#define ST32_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:42:55: error: code should be clang-formatted [-Wclang-format-violations]
#define UT16_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                      ^
librz/include/rz_types_overflow.h:42:74: error: code should be clang-formatted [-Wclang-format-violations]
#define UT16_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:42:87: error: code should be clang-formatted [-Wclang-format-violations]
#define UT16_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:43:55: error: code should be clang-formatted [-Wclang-format-violations]
#define ST16_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                      ^
librz/include/rz_types_overflow.h:43:74: error: code should be clang-formatted [-Wclang-format-violations]
#define ST16_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:43:87: error: code should be clang-formatted [-Wclang-format-violations]
#define ST16_SUB_OVFCHK(a, b) __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:44:55: error: code should be clang-formatted [-Wclang-format-violations]
#define UT8_SUB_OVFCHK(a, b)  __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                      ^
librz/include/rz_types_overflow.h:44:74: error: code should be clang-formatted [-Wclang-format-violations]
#define UT8_SUB_OVFCHK(a, b)  __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:44:87: error: code should be clang-formatted [-Wclang-format-violations]
#define UT8_SUB_OVFCHK(a, b)  __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:45:55: error: code should be clang-formatted [-Wclang-format-violations]
#define ST8_SUB_OVFCHK(a, b)  __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                      ^
librz/include/rz_types_overflow.h:45:74: error: code should be clang-formatted [-Wclang-format-violations]
#define ST8_SUB_OVFCHK(a, b)  __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                         ^
librz/include/rz_types_overflow.h:45:87: error: code should be clang-formatted [-Wclang-format-violations]
#define ST8_SUB_OVFCHK(a, b)  __builtin_sub_overflow_p (a, b, (__typeof__ ((a) - (b))) 0)
                                                                                      ^
librz/include/rz_types_overflow.h:63:3: error: code should be clang-formatted [-Wclang-format-violations]
        }
         ^
librz/include/rz_types_overflow.h:69:52: error: code should be clang-formatted [-Wclang-format-violations]
                return __builtin_mul_overflow_p(a, b, (__typeof__ ((a) * (b))) 0); \
                                                                 ^
librz/include/rz_types_overflow.h:69:65: error: code should be clang-formatted [-Wclang-format-violations]
                return __builtin_mul_overflow_p(a, b, (__typeof__ ((a) * (b))) 0); \
                                                                              ^
librz/include/rz_types_overflow.h:73:62: error: code should be clang-formatted [-Wclang-format-violations]
        static inline bool overflow_name(type_base a, type_base b) { 
                                                                    ^
librz/include/rz_types_overflow.h:74:15: error: code should be clang-formatted [-Wclang-format-violations]
                if (a > 0) { \
                            ^
librz/include/rz_types_overflow.h:75:16: error: code should be clang-formatted [-Wclang-format-violations]
                        if (b > 0) { \
                                    ^
librz/include/rz_types_overflow.h:76:29: error: code should be clang-formatted [-Wclang-format-violations]
                                return a > type_max / b; \
                                                        ^
librz/include/rz_types_overflow.h:77:5: error: code should be clang-formatted [-Wclang-format-violations]
                        } \
                         ^
librz/include/rz_types_overflow.h:78:28: error: code should be clang-formatted [-Wclang-format-violations]
                        return b < type_min / a; \
                                                ^
librz/include/rz_types_overflow.h:79:4: error: code should be clang-formatted [-Wclang-format-violations]
                } \
                 ^
librz/include/rz_types_overflow.h:80:15: error: code should be clang-formatted [-Wclang-format-violations]
                if (b > 0) { \
                            ^
librz/include/rz_types_overflow.h:81:28: error: code should be clang-formatted [-Wclang-format-violations]
                        return a < type_min / b; \
                                                ^
librz/include/rz_types_overflow.h:82:4: error: code should be clang-formatted [-Wclang-format-violations]
                } \
                 ^
librz/include/rz_types_overflow.h:83:32: error: code should be clang-formatted [-Wclang-format-violations]
                return a && b < type_max / a; \
                                             ^
```

**Comment by XVilka** [comment]
Also one test fails:
```
 60/102 rizin:unit / ovf                  FAIL            0.02s   exit status 1
```
See `test/unit/test_ovf.c`

**Review by wargio — APPROVED** [review]

File: librz/include/rz_types_overflow.h [hunk 7]
```diff
@@ -2,12 +2,23 @@
 #define RZ_TYPES_OVERFLOW_H
 
 #include <rz_types.h>
+#include <rz_userconf.h>
 
-// TODO: Use CLANG/GCC builtins if available: __builtin_mul_overflow
+// Using compiler builtins when available
 
-// ADD
```
Comment (XVilka) [inline librz/include/rz_types_overflow.h:8]: By the way, keep that `// ADD` line, just like there are `// MUL`, etc

File: librz/include/rz_userconf.h.in [hunk 8]
```diff
@@ -63,6 +63,10 @@
 #define HAVE_HEADER_SYS_MMAN_H      @HAVE_HEADER_SYS_MMAN_H@
 #define HAVE_HEADER_INTTYPES_H      @HAVE_HEADER_INTTYPES_H@
 
+#define HAVE___BUILTIN_ADD_OVERFLOW     @HAVE___BUILTIN_ADD_OVERFLOW_P@
```
Comment (XVilka) [inline librz/include/rz_userconf.h.in:66]: I recommend moving them up, together with other builtins, e.g. after `HAVE__BUILTIN_CLZLL`

**Comment by XVilka** [comment]
@KBhushan07, have you made any progress with those? Please also rebase next time you make changes since we pushed recent changes that should make CI more stable.

**Comment by XVilka** [comment]
Ping?

**Comment by XVilka** [comment]
Closed in favor of https://github.com/rizinorg/rizin/pull/4447
